core/templates/dev/head/pages/admin/AdminPageDirective.ts

```diff
@@ -93,12 +94,17 @@ oppia.directive('adminPage', ['UrlInterpolationService',
         '/pages/admin/admin_page_directive.html'),
       controllerAs: '$ctrl',
       controller: [
-        '$http', '$location', '$scope', 'AdminRouterService', 'DEV_MODE',
-        function($http, $location, $scope, AdminRouterService, DEV_MODE) {
+        '$http', '$location', '$scope', 'AdminRouterService', 'CsrfService',
+        'DEV_MODE',
+        function(
+            $http, $location, $scope, AdminRouterService, CsrfService,
+            DEV_MODE) {
           var ctrl = this;
           ctrl.userEmail = GLOBALS.USER_EMAIL;
           ctrl.inDevMode = DEV_MODE;
-
+          CsrfService.getToken().then(function(token) {
+            constants.CSRF_TOKEN = token;
```

Comment:
Don't save `CSRF_TOKEN` into `constants` just cache it inside variable in `CsrfService` and then use it everywhere where you need it.

Reply:
I tried this but due had some circular dependency issue.

Reply:
What exactly was the error? Because we don't `require()` anything in the `CsrfService`